test/secrets/BUILD

```diff
@@ -1,10 +1,29 @@
 # Test on using secrets. Obviously it can't test any real secrets
 # but it proves the system works.
 gentest(
-    name = "secrets_test",
-    outs = ["secrets_test.sh"],
+    name = "test_root_file",
+    outs = ["test_root_file.sh"],
     cmd = "echo \"#!$SECRETS\nexec true\" > $OUT",
     no_test_output = True,
     secrets = ["/bin/sh"],
     test_cmd = "$TEST",
 )
+
+tildes_replaced_test = r'''
+echo "[ \"$SECRETS_AD_SECRETS\" = \"/bin/sh $SECRETS_HOME\" ] || \
+(echo "Failed: $SECRETS_HOME $SECRETS_AD_SECRETS"; exit 1)" > $OUT'''
+
+gentest(
+    name = "test_tildes_replaced",
+    outs = ["test_tildes_replaced.sh"],
+    cmd = tildes_replaced_test,
+    no_test_output = True,
+    secrets = {
+        "home": ["~"],
```

Comment:
Not sure we should do this in a test; it will end up needing to hash the entire home directory, which could get very large.
I see the motivation though; not sure what we can add instead that we'd be sure was going to be there.

Reply:
> Thanks, looks good, just concerned about what the behaviour of the test will be!

Yeah, I originally had this as `~/.bashrc`, but even that wasn't guaranteed to exist on CirrusCI haha...If you'd like, I can remove the test for now, and have a bit less coverage?

Reply:
We could just add a sneaky `touch ~/.bashrc` in the `.cirrus.yml`? Would be nice to keep this test case.

Reply:
Done, with tests passing 🎉